lib/dep-graph-builders/npm-lock-v2/index.ts

```diff
@@ -379,3 +401,34 @@ export const getChildNodeKey = (
 
   return undefined;
 };
+const checkOverrides = (
+  ancestry: Ancestry[],
+  overrides: Overrides,
+): string | undefined => {
+  const ancestryWithoutRoot = ancestry.slice(1);
+  // First traverse into overrides from root down
+  for (const [idx, pkg] of ancestryWithoutRoot.entries()) {
+    // Do we have this in overrides
+    const value =
+      overrides[pkg.name] || overrides[`${pkg.name}@${pkg.version}`];
+
+    if (value) {
+      if (typeof value !== 'string') {
+        // Check for dotted resolution
+        if (idx + 1 === ancestryWithoutRoot.length && value['.']) {
+          return value['.'];
+        }
+
+        const res = checkOverrides(ancestryWithoutRoot, value);
+        if (res) {
+          return res;
+        }
+        continue;
+      } else {
+        return value;
+      }
```

Comment:
Reverse logic and return early

```suggestion
        if (!res) {
          continue;
        }
        return res;
      }
      return value;
      }
```